### ISSUE TITLE
[Spotify Player] Shows nothing is playing in the menu bar command

### ISSUE THREAD

**Issue by Ek2100** [raycast/extensions#7688]
### Extension

https://www.raycast.com/mattisssa/spotify-player

### Description

As seen in the image below, I clearly have a song playing however the extension is showing that nothing is playing:

<img width="1440" alt="image" src="https://github.com/raycast/extensions/assets/113011003/82c646dd-d4dd-4074-80ea-dac9ac7a2481">

### Steps To Reproduce

1. Open spotify
2. Play a song
3. Check the extension

**Comment by alessandro-newzoo** [comment]
same here, but it's on and off. Sometimes it shows nothing is playing, then sometimes it suddenly shows up again.

**Comment by peduarte** [comment]
Thank you for reporting this. There can be a delay of up to 20 seconds to synchronize what's playing in Spotify to the Menu Bar. Just to confirm, are you saying that the Menu Bar doesn't show what's playing for longer than 20 seconds?

**Comment by Ek2100** [comment]
> Thank you for reporting this. There can be a delay of up to 20 seconds to synchronize what's playing in Spotify to the Menu Bar. Just to confirm, are you saying that the Menu Bar doesn't show what's playing for longer than 20 seconds?

Yes, this was the issue. I seem to have fixed it by completely reinstalling spotify. Also, I seem to be encountering a strange issue where the play/pause command is saying is premium only

**Comment by peduarte** [comment]
Gotcha, there's an issue for that #7730 👀

**Comment by alessandro-newzoo** [comment]
> Thank you for reporting this. There can be a delay of up to 20 seconds to synchronize what's playing in Spotify to the Menu Bar. Just to confirm, are you saying that the Menu Bar doesn't show what's playing for longer than 20 seconds?

With the same track playing, if I stare at where the title should be, I see it going from this:

<img width="661" alt="image" src="https://github.com/raycast/extensions/assets/47320294/450dc494-b153-4957-859c-6d5fdabfbb58">

to just showing the Spotify logo, and back again to showing what's actually playing, and then back to just the icon.. and so on.

The amount of time between these two states seems pretty random, too.

**Comment by peduarte** [comment]
Oh that's definitely not the desired behaviour. Thanks for the extra info 🙏 Im gonna try and reproduce this, but FWIW, I've never seen this before and I've been using this extension daily for months. 

I know this isn't idea, but would you mind uninstalling the extension and installing it again? I wonder if it could be a problem due to the upgrade from the previous version, and its related to the Auth getting disconnected/reconnected. 

Thank you!

**Comment by Ek2100** [comment]
> Oh that's definitely not the desired behaviour. Thanks for the extra info 🙏 Im gonna try and reproduce this, but FWIW, I've never seen this before and I've been using this extension daily for months.
> 
> I know this isn't idea, but would you mind uninstalling the extension and installing it again? I wonder if it could be a problem due to the upgrade from the previous version, and its related to the Auth getting disconnected/reconnected.
> 
> Thank you!

I did try that however the way I fixed it was by completely reinstalling Spotify.

**Comment by peduarte** [comment]
Thank you both so much for helping 🙏 It's so weird that re-installing Spotify worked for @Ek2100 but not for you @alessandro-newzoo 🤔 

I'll have to do some digging!

**Comment by alessandro-newzoo** [comment]
thanks @peduarte !
I'd like to ask @Ek2100 to confirm whether that's still the case for them, or was it just a temporary coincidence?

**Comment by Ek2100** [comment]
> thanks @peduarte ! I'd like to ask @Ek2100 to confirm whether that's still the case for them, or was it just a temporary coincidence?

the issue is fixed for me 🙏

**Comment by michaelwexler** [comment]
I am experiencing a variation of this same issue, please see my bug at https://github.com/raycast/extensions/issues/8295. 

This is a fresh Mac, all set up on July 25, 2023.

**Comment by alessandro-newzoo** [comment]
I'm still facing the issue by the way :(
I have two MacBook M1 Pro, one with Sonoma and one with Ventura.

The extension works as expected on the Sonoma one, but I'm still having the same issue on the Ventura one (though I highly doubt the OS version has anything to do with this).

Raycast is cloud-synced on the  two machines so the extensions I have installed are exactly the same. I'm really not sure what's causing this but I would really love for it to work again as it's super handy!

**Comment by cutitdown** [comment]
Having the same issue on Monterey here! I haven't been able to get it to work it all, it's never shown the song playing in the menu bar for me. My best guess is some sort of link association is goofed because the Open Spotify button in the menu bar command tries to open the web player instead of desktop.

**Comment by peduarte** [comment]
I left a comment in https://github.com/raycast/extensions/issues/8295, will try and look into this as well as they're probably related.

**Comment by michaelwexler** [comment]
To avoid the stale moniker, has any progress been made on this one?

**Comment by peduarte** [comment]
Replied [here](https://github.com/raycast/extensions/issues/8295#issuecomment-1917194189). Will close in favour of that one to keep things clean.

**Comment by alessandro-newzoo** [comment]
> Replied [here](https://github.com/raycast/extensions/issues/8295#issuecomment-1917194189). Will close in favour of that one to keep things clean.

I believe the issues are not related. 
The one you linked points out the song info not showing up after you pause/resume, whereas this issue is to discuss the info disappearing at any time, without ever pausing Spotify.
For instance, I just started playing Spotify a few minutes ago without ever pausing, and the extension is switching to "not playing" every couple of minutes or so, then randomly comes back, and goes to not playing again etc.

**Comment by peduarte** [comment]
Thanks for the clarification. 

How are you playing it? From the Mac App, web app, iPhone?

**Comment by alessandro-newzoo** [comment]
@peduarte no worries, thanks for reopening!
I'm playing it from the Mac app (silicon). I've already tried removing and reinstalling both Spotify and the extension, but to no avail.

M1 Pro on Sonoma `14.2.1`, though the issue was already there on Ventura 6 months ago.

Happy to provide more info and logs if you tell me where to find them :)

**Comment by thomaslombart** [comment]
**This issue is now a bounty target** 💰

Hey there, we need help from our community to tackle this issue. **Working on it and having it fixed will allow you to get a month of Raycast Pro or a [mug](https://merch.raycast.com/product/mug-black).**

If you want to work on it, please let us know by posting a comment here with a rough estimate of how much time it'll take you to fix it. There are no hard deadlines, but we would like the issues to be addressed in a timely manner as this could be interesting for other contributors. You'll then be assigned the issue.

I'll be the point of contact for that. If you need more information, you can send me a message in the [Slack community](https://raycast.com/community), or in this issue. When creating your PR, please tag me either in the description or in a comment so that I can properly review it: @thomaslombart. 

I'll get in touch with you to give you the bounty once the issue is fixed.

Here's our guide about contributing to extensions: [Contribute to an extension](https://developers.raycast.com/basics/contribute-to-an-extension)

Looking forward to your contribution!

**Comment by jatindotdev** [comment]
@thomaslombart i would love to take up this task

If you assign me this issue ill start on it by tomorrow and will provide you with the updates within a week to share my findings.

**Comment by thomaslombart** [comment]
Sounds good @jatindotdev, done!

**Comment by jatindotdev** [comment]
hey @thomaslombart 

It seems that the issue lies in the fact that Spotify is not being included in the list of applications returned by the getApplications API in Raycast.

I might have to dig a little further to know why it isn’t getting detected if you can give me any reasons why this would happen

**Comment by jatindotdev** [comment]
In addition to that, since now support switching between multiple clients we can just make the menubar command not rely on the mac app and show the now playing data from the api, even now we do it we fetch a lot of data which can be used without any loss

**Comment by Joshua-Wilcox** [comment]
I also have this issue.. the "Current Track" command works just fine but the menu bar never appears :(

- separately, I also have to **keep on** logging back into spotify on raycast - as in like 5x a day

**Comment by erics118** [comment]
@raycastbot keep this issue open

**Comment by kennguyen107** [comment]
I'm also having this issue. Is there any fix yet for it?

**Comment by d-makogon** [comment]
I had an unrelated issues, and to fix that I ran 
`sudo killall coreaudiod`

All of the sudden, the spotify extension came back to life for me.

**Comment by Durthander** [comment]
Hi! 👋
I’m experiencing the same issue described.
The Spotify Player extension works perfectly for all commands (play, pause, skip, search, etc.), but the Menu Bar Player always shows “Nothing is playing” — even when a song is actively playing on my Mac Spotify app.

**Comment by aadesouz** [comment]
Also experiencing this issue, the menu bar item doesn't appear or populate at all despite functioning fine within the Raycast app
